[Update] Article Skip links best practices (#601)
Orange-OpenSource/a11y-guidelines: src/en/articles/skip-links-best-practices.md

We can distinguish 3 types of skip links:

1. **quick access links:** gathered at the top of the page to go to the main regions of the page: "Skip to menu", "Skip to content", "Skip to search", for example

> 2. **section skip links:** positioned before the region that they allow to jump: "Skip the section", "Skip the chapter"

Reviewer: ```suggestion
2. **section skip links:** positioned before the region that they allow to skip: "Skip the section", "Skip the chapter"
```

- Users who are struggling to navigate a large page: motor impaired people (fatigability or motor limitations) or people on their smartphone (this is a way to spare the user from swiping to browse the page)
- the visually impaired, who use or not a screen magnifier, and have trouble having a global representation of the topology of the page

> We will therefore, thanks to skip links, limit the use of keyboard navigation keys and more generally save actions during intra-page navigation.

Reviewer: ```suggestion
We will therefore, thanks to skip links, limit the use of keyboard navigation keys and more generally decrease the gestures to be done when navigating between pages.
```

## Use

It is generally a link pointing to an HTML element with an id attribute.

```html
<body>
  <a class="skip" href="#content">Go to content</a>
  […]
  <main id="content">
     […]
  </main>
  […]
</body>
```

## What are the best practices?

> - It is possible to integrate a skip link in the form of an image (such as an 'arrow' with a <code>title</code> attribute) that appears after scrolling down the page. The skip link will allow for a direct return to the top of the page. This avoidance link should not hinder the reading or understanding of the information; it should be the last keyboard-focusable element.

Reviewer: ```suggestion
- It is possible to embed a skip link as an image (such as an 'arrow' with a <code>title</code> attribute) that appears after scrolling the page. The skip link returns directly to the top of the page. This avoidance link should not hinder the reading or understanding of the information; it should be the last keyboard-focusable element.
```

- The skip link must be reachable by keyboard navigation and regardless of the direction of navigation.(<kbd>TAB</kbd> ou <kbd>Shift</kbd> + <kbd>TAB</kbd>).
- Placing a skip link on an <code>id</code> works, but targeting the skip link on elements like <code>aside</code>, <code>footer</code>, or <code>main</code> makes it less sensitive to potential changes (such as an <code>id</code> change, or simply not being included in the code of a new page, for instance).

> - Skip or quick access links should be visually located in the same place on the page and in the same relative order in the source code across all pages of the site.

Author: Not sure about this one, for me 'across' is better here. 🤔

> - Skip or quick access links should be visually located in the same place on the page and in the same relative order in the source code across all pages of the site.

Reviewer: good for me

- navigation contains a lot of links
- the content has a lot of links (several navigation menus, footers acting as a site map, etc.)
- the page is divided into many different parts (portal, dashboard, etc.)
- there is no other way to navigate within the page (section title, HTML5 semantic structure…)

> **Note**: keep in mind that for a skip link to be functional, it should not merely scroll the page to the indicated location (such as the main content). It must allow the user to 'skip' a part of the page. If a user activates a 'Go to content' link using the keyboard, on the next <kbd>TAB</kbd>, the focus should move to the main content and not to the next avoidance link. This focus can be achieved by placing an anchor to the next <code>id</code> to target, for example."

Author: ```suggestion
**Note**: keep in mind that for a skip link to be functional, it should not simply scroll to the targeted location (such as the main content for instance). It must allow the user to 'skip' a part of the page. If a user activates a 'Go to content' link using the keyboard, on the next <kbd>TAB</kbd>, the focus should move to the main content and not to the next avoidance link. This focus can be achieved by placing an anchor to the next <code>id</code> to target, for example."
```

> **Note**: keep in mind that for a skip link to be functional, it should not merely scroll the page to the indicated location (such as the main content). It must allow the user to 'skip' a part of the page. If a user activates a 'Go to content' link using the keyboard, on the next <kbd>TAB</kbd>, the focus should move to the main content and not to the next avoidance link. This focus can be achieved by placing an anchor to the next <code>id</code> to target, for example."

Author: 'Simply' is better, thanks.
But I am not ok with 'main location' since it only works it the skip link allows user to go to the main. I changed it for 'targeted location (such as the main content for instance)', is it ok for you @damiengomez?
